src/lib.rs

```diff
@@ -133,87 +120,29 @@ impl fmt::Display for Error {
     }
 }
 
-/// The term of a log entry.
-#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
-pub struct Term(u64);
-impl From<u64> for Term {
-    fn from(val: u64) -> Term {
-        Term(val)
-    }
-}
-impl Into<u64> for Term {
-    fn into(self) -> u64 {
-        self.0
-    }
-}
-impl ops::Add<u64> for Term {
-    type Output = Term;
-    fn add(self, rhs: u64) -> Term {
-        Term(self.0.checked_add(rhs).expect("overflow while incrementing Term"))
-    }
-}
-impl ops::Sub<u64> for Term {
-    type Output = Term;
-    fn sub(self, rhs: u64) -> Term {
-        Term(self.0.checked_sub(rhs).expect("underflow while decrementing Term"))
-    }
-}
-impl fmt::Display for Term {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        fmt::Display::fmt(&self.0, f)
-    }
-}
+/// The Term for the log entry.
+pub type Term = u64;
```

Comment:
I'd prefer not to make this change.  A typedef is not the same as a newtype, and the latter has better type checking (specifically, it's impossible to pass a Term where a LogIndex should go, and vice versa).